impl/src/as_ref.rs

```diff
@@ -79,3 +76,193 @@ pub fn expand(input: &DeriveInput, trait_name: &'static str) -> Result<TokenStre
         }
     )*})
 }
+
+enum StructAttribute {
+    Forward,
+}
+
+impl StructAttribute {
+    fn parse_attrs(attrs: impl AsRef<[syn::Attribute]>) -> syn::Result<Option<Self>> {
+        attrs
+            .as_ref()
+            .iter()
+            .filter(|attr| attr.path().is_ident("as_ref"))
+            .try_fold(None, |mut attrs, attr| {
+                let field_attr = attr.parse_args()?;
+                if attrs.replace(field_attr).is_some() {
+                    Err(syn::Error::new(
+                        attr.path().span(),
+                        "only single `#[as_ref(...)]` attribute is allowed here",
```

Comment:
can you add some basic `compile_fail` tests for the errors you're introducing here?